bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/GarbageCollectorThread.java

```diff
@@ -470,6 +473,10 @@ public void runWithFlags(boolean force, boolean suspendMajor, boolean suspendMin
                     gcStats.getMajorCompactionCounter().inc();
                     majorCompacting.set(false);
                 }
+                if (entryLocationCompaction) {
```

Comment:
I suggest moving the entryLocationCompact out of the major or minor compaction scope and using a separate interval to control it.  Remove the `entryLocationCompaction` flag and use the entryLocationCompactionInterval instead. The default value is `-1`, which means disable. We can set it to 1 week by default in the future after verified this operation doesn't introduce a huge impact on performance.

The major or minor compaction has many impact factors, such as suspension, force, and interval.  It will make the entryLocation  compaction behavior unpredictable

Reply:
Great suggestion! Currently, the execution of `entryLocationCompact` is indeed coupled with the triggering of `majorCompaction`. I will modify it so that `entryLocationCompact` execution is only based on the configured trigger interval.